src/sentry/services/hybrid_cloud/filter_query.py

```diff
@@ -21,71 +21,50 @@
 OpaqueSerializedResponse = Any
 
 
-# A collection of classes that can be used to quickly provision interfaces & implementations for basic RPC Hybrid Cloud service implementations.
+# A class that can be used to quickly provision interfaces & implementations for basic RPC Hybrid Cloud service implementations.
 # It enforces several niceties:
 #   * Type safety at call sites: the public API can be inferred, unknown arguments are rejected,
 #     all filter arguments are optional, and nullable filters are explicitly Optional[...]
 #   * Filter argument validation: an arbitrary function can be used to ensure that proper filters are applied
 #     e.g. we don't want to just query for every row if no filters are passed
 #   * A standard interface across most of our simple rpc services
-
-# This class should be inherited from to add this functionality to the public interface for a service.
-# E.g. class UserService(FilterQueryInterface[
-#        UserQueryArgs, RpcUser, Union[UserSerializerResponse, UserSerializerResponseSelf]
-#      ], InterfaceWithLifecycle):
-#         ...
-class FilterQueryInterface(Generic[FILTER_ARGS, RPC_RESPONSE, SERIALIZER_ENUM], abc.ABC):
-    @abc.abstractmethod
-    def serialize_many(
-        self,
-        *,
-        filter: FILTER_ARGS,
-        as_user: Optional[RpcUser] = None,
-        auth_context: Optional[AuthenticationContext] = None,
-        serializer: Optional[SERIALIZER_ENUM] = None,
-    ) -> List[OpaqueSerializedResponse]:
-        pass
-
-    @abc.abstractmethod
-    def get_many(self, *, filter: FILTER_ARGS) -> List[RPC_RESPONSE]:
-        pass
-
-
-# This class should be inherited from to add this functionality to the database implementation for a service.
-# E.g. class DatabaseBackedUserService(
-#         FilterQueryDatabaseImpl[User, UserFilterArgs, RpcUser, UserSerializeType],
-#         UserService,
-#      ):
-#       ...
+#
+# This class should be delegated to, to add this functionality to the public interface for a service.
+# E.g.
+# class DatabaseBackedUserService(UserService):
+#     def serialize_many(...) -> List[OpaqueSerializedResponse]:
+#         return self._UserFilterQuery().serialize_many(filter, as_user, auth_context, serializer)
+#
+#    class _UserFilterQuery(FilterQueryDatabaseImpl[User, UserFilterArgs, RpcUser, UserSerializeType]): ...
 class FilterQueryDatabaseImpl(
     Generic[BASE_MODEL, FILTER_ARGS, RPC_RESPONSE, SERIALIZER_ENUM], abc.ABC
 ):
     # Required Overrides
 
     @abc.abstractmethod
-    def _base_query(self) -> QuerySet:
+    def base_query(self) -> QuerySet:
```

Comment:
Made these methods non-private because `DatabaseBackedUserService` wants to call several of them directly.